service/mapper/transitions_internal_test.go

```diff
@@ -73,32 +76,17 @@ func TestNewTransitions(t *testing.T) {
 
 func TestTransitions_BootstrapState(t *testing.T) {
 	t.Run("nominal case", func(t *testing.T) {
-		// t.Parallel()
-		t.Skip("fix later")
-
-		tr, st := baselineFSM(t, StatusBootstrap)
-
-		// Copy state in local scope so that we can override its SaveFunc without impacting other
-		// tests running in parallel.
-		var saveCalled bool
-		forest := mocks.BaselineForest(t, true)
-		forest.SaveFunc = func(tree *trie.MTrie, paths []ledger.Path, parent flow.StateCommitment) {
-			if !saveCalled {
-				assert.True(t, tree.IsEmpty())
-				assert.Nil(t, paths)
-				assert.Zero(t, parent)
-				saveCalled = true
-				return
-			}
-			assert.False(t, tree.IsEmpty())
-			assert.Len(t, tree.AllPayloads(), len(paths))
-			assert.Len(t, paths, 3) // Expect the three paths from leaves.
-			assert.NotZero(t, parent)
-		}
-
-		err := tr.BootstrapState(st)
-		assert.NoError(t, err)
-		assert.Len(t, st.registers, 3) // same number of registers as paths
+		t.Parallel()
+		unittest.RunWithTempDir(t, func(dir string) {
+			logger := unittest.Logger()
+			tries := createSimpleTrie(t)
+			fileName := "test_checkpoint_file"
+			require.NoErrorf(t, wal.StoreCheckpointV6Concurrently(tries, dir, fileName, &logger), "fail to store checkpoint")
+			tr, st := baselineFSM(t, StatusBootstrap)
+			st.checkpointFileName = fileName
+			st.checkpointDir = dir
+			require.NoError(t, tr.BootstrapState(st))
```

Comment:
can we also validate that the last height is updated?

Reply:
Done!